tests/helpers.py

```diff
@@ -56,3 +56,15 @@ def trigger(
         if post_event:
             post_event(mgr.charm)
     return state_out
+
+
+def sort_patch(patch: List[Dict], key=lambda obj: obj["path"] + obj["op"]):
```

Comment:
I think that jsonpatch would generate at most one entry for each key and `obj["op"]` is not necessary.

The available ops are: add, remove, replace.

If it is somehow necessary, consider `(obj["path"], obj["op"])` instead.

Reply:
This method isn't new in this PR by the way, it looks like it is because it somehow got lost in a merge, but once everything is rebased, it'll vanish from this PR.